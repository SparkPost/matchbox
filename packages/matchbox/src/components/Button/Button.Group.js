import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './Button.module.scss';

const Group = ({ children, className = '' }) => (
  <div className={classnames(styles.Group, className)}>
    {children}
  </div>
);

Group.propTypes = {
  children: PropTypes.node.isRequired
};

Group.displayName = 'Button.Group';

export default Group;
