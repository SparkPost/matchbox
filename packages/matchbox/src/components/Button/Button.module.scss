@import '../../styles/config';

.Button,
.Button:visited {
  display: inline-block;
  padding: rem(5) rem(12);
  border: none;
  margin-bottom: 0;

  border-radius: border-radius();

  font-size: font-size(400);
  font-weight: 500;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  transition-property: background, color, border, box-shadow, outline;
  transition-duration: .15s;

  text-decoration: none;
  white-space: nowrap;
  line-height: rem(24);

  &:last-child {
    margin-bottom: 0;
  }

  .Group > * &, .Group > &{
    margin-right: -1px;
    border-radius: 0;

    &:first-child.Button:not(.flat) {
      border-top-left-radius: border-radius();
      border-bottom-left-radius: border-radius();
    }

    &:last-child.Button:not(.flat) {
      border-top-right-radius: border-radius();
      border-bottom-right-radius: border-radius();
    }
  }

  // Default button styling
  background: linear-gradient(180deg, color(gray, 10), color(gray, 9));
  border: 1px solid color(gray, 7);
  color: color(gray, 2);
  box-shadow: shadow();

  &:hover {
    background: linear-gradient(180deg, color(gray, 9), color(gray, 9));
    color: color(gray, 2);
  }

  &:hover:focus {
    box-shadow: shadow(),
                inset 0 3px 4px 0 rgba(color(gray, 2), 0.1);
  }

  &:focus {
    position: relative;
    z-index: 1;
    outline: 0;
    border: 1px solid color(blue);
    box-shadow: 0 0 0 1px color(blue);
  }

  &.disabled {
    opacity: 0.6;
    background: color(gray, 9);
    &:hover {
      background: color(gray, 9);
      cursor: not-allowed;
    }
  }
}

// Color Styles
.color-orange,
.color-orange:visited {
  background: linear-gradient(180deg, color(orange), color(orange, dark));
  color: color(gray, 10);
  border: 1px solid darken(color(orange), 12);

  &:hover {
    color: color(gray, 10);
    background: linear-gradient(180deg, darken(color(orange, dark), 3), darken(color(orange, dark), 3));
  }

  &:focus {
    border: 1px solid darken(color(orange), 12);
    box-shadow: inset 0 3px 4px 0 rgba(color(gray, 2), 0.1),
                0 0 0 2px rgba(color(orange), 0.6);
  }

  &.disabled {
    background: linear-gradient(180deg, color(orange), color(orange, dark));
    &:hover {
      background: linear-gradient(180deg, color(orange), color(orange, dark));
    }
  }

  &.flat {
    color: color(orange);
    &:hover, &:active, &:focus { color: color(orange); }
  }
}

.color-blue,
.color-blue:visited {
  background: linear-gradient(180deg, color(blue), color(blue, dark));
  color: color(gray, 10);
  border: 1px solid darken(color(blue), 12);

  &:hover {
    color: color(gray, 10);
    background: linear-gradient(180deg, darken(color(blue, dark), 3), darken(color(blue, dark), 3));
  }

  &:focus {
    border: 1px solid darken(color(blue), 12);
    box-shadow: inset 0 3px 4px 0 rgba(color(gray, 2), 0.1),
                0 0 0 2px rgba(color(blue), 0.6);
  }

  &.disabled {
    background: linear-gradient(180deg, color(blue), color(blue, dark));
    &:hover {
      background: linear-gradient(180deg, color(blue), color(blue, dark));
    }
  }

  &.flat {
    color: color(blue, dark);
    &:hover, &:active, &:focus { color: color(blue, dark); }
  }
}

.color-navy,
.color-navy:visited {
  background: linear-gradient(180deg, color(navy), color(navy, dark));
  color: color(gray, 10);
  border: 1px solid darken(color(navy), 12);

  &:hover {
    color: color(gray, 10);
    background: linear-gradient(180deg, darken(color(navy, dark), 3), darken(color(navy, dark), 3));
  }

  &:focus {
    border: 1px solid darken(color(navy), 20);
    box-shadow: inset 0 3px 4px 0 rgba(color(gray, 2), 0.1),
                0 0 0 2px color(navy, light);
  }

  &.disabled {
    background: linear-gradient(180deg, color(navy), color(navy, dark));
    &:hover {
      background: linear-gradient(180deg, color(navy), color(navy, dark));
    }
  }

  &.flat {
    color: color(navy);
    &:hover, &:active, &:focus { color: color(navy); }
  }
}

.color-red,
.color-red:visited {
  background: linear-gradient(180deg, color(red), color(red, dark));
  color: color(gray, 10);
  border: 1px solid color(red, dark);

  &:hover {
    color: color(gray, 10);
    background: linear-gradient(180deg, darken(color(red, dark), 3), darken(color(red, dark), 3));
  }

  &:focus {
    border: 1px solid color(red, dark);
    box-shadow: inset 0 3px 4px 0 rgba(color(gray, 2), 0.2),
                0 0 0 2px rgba(color(red, dark), 0.6);
  }

  &.disabled {
    background: linear-gradient(180deg, color(red), color(red, dark));
    &:hover {
      background: linear-gradient(180deg, color(red), color(red, dark));
    }
  }

  &.flat {
    color: color(red);
    &:hover, &:active, &:focus { color: color(red); }
  }
}

.color-purple,
.color-purple:visited {
  background: linear-gradient(180deg, color(purple), color(purple, dark));
  color: color(gray, 10);
  border: 1px solid color(purple, dark);

  &:hover {
    color: color(gray, 10);
    background: linear-gradient(180deg, darken(color(purple, dark), 3), darken(color(purple, dark), 3));
  }

  &:focus {
    border: 1px solid color(purple, dark);
    box-shadow: inset 0 3px 4px 0 rgba(color(gray, 2), 0.2),
                0 0 0 2px rgba(color(purple, dark), 0.6);
  }

  &.disabled {
    background: linear-gradient(180deg, color(purple), color(purple, dark));
    &:hover {
      background: linear-gradient(180deg, color(purple), color(purple, dark));
    }
  }

  &.flat {
    color: color(purple);
    &:hover, &:active, &:focus { color: color(purple); }
  }
}

// Plain Styles
.flat,
.flat:visited {
  background: transparent;
  box-shadow: none;
  border: none;
  border-radius: 0;
  color: color(gray, 2);

  &:hover, &:active {
    color: color(gray, 1);
    background: color(gray, 9);
    box-shadow: none;
  }

  &:focus {
    color: color(gray, 1);
    box-shadow: 0 0 0 2px color(gray, 7);
    border: none;
  }

  &.disabled {
    opacity: 0.6;
    background: transparent;
    &:hover {
      background: transparent;
    }
  }
}

// Destructive Styles
.destructive,
.destructive:visited {
  color: color(red, dark);

  &:hover {
    color: color(red, dark);
  }
}

// Outline Styles
.outline,
.outline:visited {
  background: transparent;
  color: color(gray, 2) !important;
  border: 1px solid color(gray, 5);

  &:hover {
    background: rgba(color(gray, 2), 0.08);
    color: color(gray, 1);
  }

  &:focus {
    border: 1px solid color(gray, 3);
    background: rgba(color(gray, 3), 0.1);
    box-shadow: shadow(),
                0 0 0 2px color(gray, 4);
  }

  &.disabled {
    background: transparent;
    &:hover {
      background: transparent;
    }
  }
}

// Sizing
.small {
  padding: rem(3) rem(9);
  font-size: font-size(300);
}

.large {
  padding: rem(9) rem(15);
}

.fullWidth {
  display: block;
  width: 100%;
  text-align: center;
}

.Group {
  display: inline-block;
}
