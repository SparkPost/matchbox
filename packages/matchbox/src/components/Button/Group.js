import React from 'react';
import PropTypes from 'prop-types';

const Group = ({ children, className = '' }) => (
  <div className={className}>
    {children}
  </div>
);

Group.propTypes = {
  children: PropTypes.node.isRequired
};

Group.displayName = 'Button.Group';

export default Group;
