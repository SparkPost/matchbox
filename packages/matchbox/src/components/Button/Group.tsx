import React from 'react';
import styled from 'styled-components';
import { Box } from '../Box';
import { group } from './styles';

export type ButtonGroupProps = {
  className?: string;
  children?: React.ReactNode;
};

const StyledGroup = styled(Box)`
  ${group}
`;

const Group = ({ children, className = '' }: ButtonGroupProps): JSX.Element => (
  <StyledGroup display="inline-flex" alignItems="center" className={className}>
    {children}
  </StyledGroup>
);

Group.displayName = 'Button.Group';

export default Group;
