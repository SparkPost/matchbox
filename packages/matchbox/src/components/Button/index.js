export * from './Button';
export { default as Button } from './Button';
