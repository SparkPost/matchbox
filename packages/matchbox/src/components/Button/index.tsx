export * from './utils';
export { default as Button } from './Button';
