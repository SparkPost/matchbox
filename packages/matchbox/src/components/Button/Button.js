import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Group from './Group';
import styles from './Button.module.scss';

class Button extends Component {
  static displayName = 'Button';

  static Group = Group;

  static propTypes = {
    color: PropTypes.oneOf(['orange', 'blue', 'navy', 'purple', 'red']),
    disabled: PropTypes.bool,
    destructive: PropTypes.bool,
    flat: PropTypes.bool,
    outline: PropTypes.bool,
    size: PropTypes.oneOf(['small', 'large', 'default']),
    fullWidth: PropTypes.bool,
    submit: PropTypes.bool,
    to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    external: PropTypes.bool,
    component: PropTypes.oneOfType([
      PropTypes.func,
      PropTypes.element
    ]),
    children: PropTypes.node
  }

  static defaultProps = {
    size: 'default'
  }

  handleMouseUp({ currentTarget }) {
    currentTarget.blur();
  }

  render() {
    const {
      children,

      // Styles
      primary, // Deprecate in favor of color
      color,
      disabled,
      destructive,
      plain, // Deprecate in favor of flat
      flat,
      outline,

      // Options
      size,
      fullWidth,
      submit,

      to,
      Component,
      component,
      external,

      // Events
      onClick,
      onFocus,
      onBlur,

      className = '',
      ...rest
    } = this.props;

    const WrapperComponent = component || Component;

    const buttonColor = primary ? 'orange' : color;

    const classname = classnames(
      styles.Button,
      buttonColor && styles[`color-${buttonColor}`],
      disabled && styles.disabled,
      destructive && styles.destructive,
      (flat || plain) && styles.flat,
      outline && styles.outline,
      fullWidth && styles.fullWidth,
      (size && size !== 'default') && styles[`${size}`],
      className
    );

    if (to && !WrapperComponent) {
      return (
        <a
          href={to}
          target={external ? '_blank' : ''}
          rel={external ? 'noopener noreferrer' : ''}
          onClick={onClick}
          onFocus={onFocus}
          onBlur={onBlur}
          className={classname}
          disabled={disabled}
          onMouseUp={this.handleMouseUp}
          {...rest}
        >
          {children}
        </a>
      );
    }

    if (to && WrapperComponent) {
      return (
        <WrapperComponent
          to={to}
          onClick={onClick}
          onFocus={onFocus}
          onBlur={onBlur}
          className={classname}
          disabled={disabled}
          onMouseUp={this.handleMouseUp}
          {...rest}
        >
          {children}
        </WrapperComponent>
      );
    }

    return (
      <button
        type={submit ? 'submit' : 'button'}
        onClick={onClick}
        onFocus={onFocus}
        onBlur={onBlur}
        className={classname}
        disabled={disabled}
        onMouseUp={this.handleMouseUp}
        {...rest}
      >
        {children}
      </button>
    );
  }
}

export default Button;
