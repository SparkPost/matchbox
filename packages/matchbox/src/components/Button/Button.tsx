import React from 'react';
import PropTypes from 'prop-types';
import { Transition } from 'react-transition-group';
import { deprecate } from '../../helpers/propTypes';
import styled from 'styled-components';
import {
  margin,
  MarginProps,
  width,
  WidthProps,
  padding,
  PaddingProps,
  compose,
} from 'styled-system';
import { createPropTypes } from '@styled-system/prop-types';
import { omit } from '@styled-system/props';
import { pick, clean } from '../../helpers/props';
import { Box } from '../Box';
import { Spinner } from '../Spinner';
import Icon from './Icon';
import Group from './Group';
import {
  base,
  focus,
  visualSize,
  colorVariant,
  disabled,
  fullWidth,
  loader,
  childwrapper,
} from './styles';

import type * as Polymorphic from '../../helpers/types';

export function getLoaderColor({ variant = 'filled', color = 'gray' } = {}): string {
  if (variant === 'filled') {
    if (color === 'white') {
      return 'gray';
    }

    return 'white';
  }

  if (color === 'white') {
    return 'white';
  }

  if (color === 'blue') {
    return 'blue';
  }

  return 'gray';
}

// TODO Categorize system props and abstract
const system = compose(margin, width, padding);

export const StyledButton = styled(Box).withConfig(clean(['loading']))`
  ${base}
  ${focus}
  ${visualSize}
  ${colorVariant}
  ${disabled}
  ${fullWidth}
  ${system}
`;

const StyledLoader = styled.div`
  ${loader}
`;

const ChildWrapper = styled.span.withConfig(clean(['loading']))`
  ${childwrapper}
`;

// Handles deprecated button variant props
// TODO Remove in 6.0
function getVariant({ outline, outlineBorder, plain, flat, variant }): string {
  if (variant) {
    return variant;
  }

  if (outlineBorder) {
    return 'outline';
  }

  if (outline) {
    return 'mutedOutline';
  }

  if (plain || flat) {
    return 'text';
  }

  return 'filled';
}

interface BaseProps {
  /**
   * @deprecated Use `as` instead
   */
  component?: React.ElementType;
  /**
   * @deprecated Use `as` instead
   */
  Component?: React.ElementType;
  primary?: boolean;
  color?: 'red' | 'gray' | 'blue' | 'white';
  disabled?: boolean;
  /**
   * @deprecated Use `color` instead
   */
  destructive?: boolean;
  loading?: boolean;
  loadingLabel?: string;
  /**
   * @deprecated Use `variant` instead
   */
  plain?: boolean;
  /**
   * @deprecated Use `variant` instead
   */
  flat?: boolean;
  /**
   * @deprecated Use `variant` instead
   */
  outline?: boolean;
  /**
   * @deprecated Use `variant` instead
   */
  outlineBorder?: boolean;
  variant?: 'outline' | 'mutedOutline' | 'text' | 'filled';
  size?: 'default' | 'small' | 'large';
  to?: string;
  fullWidth?: boolean;
  submit?: boolean;
  external?: boolean;
  title?: string;
  children?: React.ReactNode;
}

type PolymorphicButton = Polymorphic.ForwardRefComponent<'button', BaseProps &
  MarginProps &
  WidthProps &
  PaddingProps>;

const Button = React.forwardRef(
  (props, ref) => {
    const {
      children,

      // Styles
      primary, // Deprecate in favor of color
      color,
      disabled,
      destructive, // Deprecate in favor of color
      loading,
      loadingLabel,

      // Below 3 props to be deprecated for a 'variant' prop
      plain, // Deprecate in favor of flat
      flat,
      outline,
      outlineBorder,
      variant,

      // Options
      // Renaming to prevent `width` and `height` pass through
      // Size is a valid styled-system prop
      size: buttonSize,
      fullWidth,
      submit,

      to,
      Component, // Deprecate in favor of component
      component,
      as,
      external,
      title,

      // Events
      onClick,
      onFocus,
      onBlur,

      className = '',
      ...rest // TODO remove spreading of unknown props
    } = props;

    const transitionRef = React.useRef();

    const systemProps = pick(rest, system.propNames);
    const componentProps = omit(rest);

    // Polyfills deprecrated 'Component' prop â€” use as instead of component
    const WrapperComponent = as || component || Component;

    // Polyfills to be deprecrated 'primary' and 'destructive' prop
    const buttonColor = primary ? 'blue' : destructive ? 'red' : color;

    const buttonVariant = React.useMemo(() => {
      return getVariant({ variant, outline, outlineBorder, plain, flat });
    }, [variant, outline, outlineBorder, plain, flat, getVariant]);

    const loaderColor = React.useMemo(() => {
      return getLoaderColor({ variant: buttonVariant, color });
    }, [buttonVariant, color]);

    const loadingIndicator = React.useMemo(() => {
      return (
        <Transition mountOnEnter unmountOnExit in={loading} timeout={0} nodeRef={transitionRef}>
          {state => (
            <StyledLoader state={state} ref={transitionRef}>
              <Spinner
                // TODO
                // @ts-ignore
                color={loaderColor}
                size="small"
                label={loadingLabel}
                rotationOnly
              />
            </StyledLoader>
          )}
        </Transition>
      );
    }, [loading]);

    const sharedProps = {
      className,
      disabled: disabled || loading,
      fullWidth,
      onClick,
      onFocus,
      onBlur,
      buttonSize,
      buttonVariant,
      buttonColor,
      ref,
      loading,
      ...systemProps,
      ...componentProps,
    };

    const childrenMarkup = React.useMemo(() => {
      return (
        <ChildWrapper aria-hidden={loading} loading={loading ? loading : undefined}>
          {children}
        </ChildWrapper>
      );
    }, [loading, children]);

    if (to && !WrapperComponent) {
      return (
        <StyledButton
          as="a"
          href={to}
          target={external ? '_blank' : ''}
          rel={external ? 'noopener noreferrer' : ''}
          title={external && !title ? 'Opens in a new tab' : title}
          {...sharedProps}
        >
          {childrenMarkup}
          {loadingIndicator}
        </StyledButton>
      );
    }

    if (to && WrapperComponent) {
      return (
        <StyledButton as={WrapperComponent} to={to} {...sharedProps}>
          {childrenMarkup}
          {loadingIndicator}
        </StyledButton>
      );
    }

    return (
      <StyledButton
        as={WrapperComponent || 'button'}
        type={submit ? 'submit' : 'button'}
        {...sharedProps}
      >
        {childrenMarkup}
        {loadingIndicator}
      </StyledButton>
    );
  },
) as PolymorphicButton & {
  Group: typeof Group;
  Icon: typeof Icon;
};

Button.displayName = 'Button';
Button.Group = Group;
Button.Icon = Icon;

Button.propTypes = {
  as: PropTypes.elementType,
  children: PropTypes.node,
  color: PropTypes.oneOf(['gray', 'orange', 'blue', 'navy', 'purple', 'red', 'white']),
  component: deprecate(PropTypes.elementType, 'Use `as` instead'),
  disabled: PropTypes.bool,
  external: PropTypes.bool,
  fullWidth: PropTypes.bool,
  loading: PropTypes.bool,
  loadingLabel: PropTypes.string,
  size: PropTypes.oneOf(['small', 'large', 'default']),
  submit: PropTypes.bool,
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  title: PropTypes.string,
  variant: PropTypes.oneOf(['filled', 'outline', 'text', 'mutedOutline']),

  // Deprecated props
  Component: deprecate(PropTypes.elementType, 'Use `as` instead'),
  destructive: deprecate(PropTypes.bool, 'Use the `color` prop instead'),
  flat: deprecate(PropTypes.bool, 'Use `variant` instead'),
  primary: deprecate(PropTypes.bool, 'Use `color` prop instead'),
  outline: deprecate(PropTypes.bool, 'Use `variant` instead'),
  outlineBorder: deprecate(PropTypes.bool, 'Use `variant` instead'),
  plain: deprecate(PropTypes.bool, 'Use `variant` instead'),

  // Undocumented helper function
  // https://github.com/styled-system/styled-system/issues/618
  // TODO Abstract when system props are grouped
  ...createPropTypes(margin.propNames),
  ...createPropTypes(padding.propNames),
  ...createPropTypes(width.propNames),
};

Button.defaultProps = {
  size: 'default',
  loadingLabel: 'Loading',
};

export default Button;
