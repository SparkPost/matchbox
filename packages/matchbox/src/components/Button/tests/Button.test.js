import React from 'react';
import Button from '../Button';
import { shallow } from 'enzyme';
import cases from 'jest-in-case';

describe('Button', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Button>Hola!</Button>);
  });

  const testCases = [
    { name: 'large size', props: { size: 'large' }},
    { name: 'disabled', props: { disabled: true }},
    { name: 'destructive', props: { destructive: true }},
    { name: 'flat', props: { flat: true }},
    { name: 'flat with color', props: { flat: true, color: 'red' }},
    { name: 'flat with color and disabled', props: { flat: true, color: 'red', disabled: true }},
    { name: 'color navy', props: { color: 'navy' }},
    { name: 'outline enabled', props: { outline: true }},
    { name: 'fullWidth', props: { fullWidth: true }},
    { name: 'submit', props: { submit: true }},
    { name: 'external button', props: { external: true }},
    { name: 'to without component', props: { to: '/nocomp' }},
    { name: 'external to without component', props: { to: '/nocomp', external: true }},
    { name: 'to with component', props: { to: '/withcomp', component: jest.fn }},
    { name: 'to while disabled', props: { to: '/withcomp', component: jest.fn, disabled: true }},
    { name: 'deprecated prop - primary', props: { primary: true }}
  ];

  cases('renders button states', (opts) => {
    wrapper.setProps(opts.props);
    expect(wrapper).toMatchSnapshot();
  }, testCases);

  cases('invokes event', (opts) => {
    const fn = jest.fn();
    const newProps = {};
    newProps[opts.name] = fn;
    wrapper.setProps(newProps);

    wrapper.simulate(opts.event);
    expect(fn).toHaveBeenCalledTimes(1);
  }, [
    { name: 'onClick', event: 'click' },
    { name: 'onBlur', event: 'blur' },
    { name: 'onFocus', event: 'focus' }
  ]);

  it('should handle mouse up', () => {
    const e = { currentTarget: { blur: jest.fn() }};
    wrapper.simulate('mouseUp', e);
    expect(e.currentTarget.blur).toHaveBeenCalled();
  });
});
