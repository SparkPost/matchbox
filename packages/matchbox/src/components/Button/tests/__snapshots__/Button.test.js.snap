// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button states color navy 1`] = `
<button
  className="Button color-navy"
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states deprecated prop - primary 1`] = `
<button
  className="Button color-orange"
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states destructive 1`] = `
<button
  className="Button destructive"
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states disabled 1`] = `
<button
  className="Button disabled"
  disabled={true}
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states external button 1`] = `
<button
  className="Button"
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states external to without component 1`] = `
<a
  className="Button"
  href="/nocomp"
  onMouseUp={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  Hola!
</a>
`;

exports[`Button renders button states flat 1`] = `
<button
  className="Button flat"
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states flat with color 1`] = `
<button
  className="Button color-red flat"
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states flat with color and disabled 1`] = `
<button
  className="Button color-red disabled flat"
  disabled={true}
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states fullWidth 1`] = `
<button
  className="Button fullWidth"
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states large size 1`] = `
<button
  className="Button large"
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states outline enabled 1`] = `
<button
  className="Button outline"
  onMouseUp={[Function]}
  type="button"
>
  Hola!
</button>
`;

exports[`Button renders button states submit 1`] = `
<button
  className="Button"
  onMouseUp={[Function]}
  type="submit"
>
  Hola!
</button>
`;

exports[`Button renders button states to while disabled 1`] = `
<bound fn
  className="Button disabled"
  disabled={true}
  onMouseUp={[Function]}
  to="/withcomp"
>
  Hola!
</bound fn>
`;

exports[`Button renders button states to with component 1`] = `
<bound fn
  className="Button"
  onMouseUp={[Function]}
  to="/withcomp"
>
  Hola!
</bound fn>
`;

exports[`Button renders button states to without component 1`] = `
<a
  className="Button"
  href="/nocomp"
  onMouseUp={[Function]}
  rel=""
  target=""
>
  Hola!
</a>
`;
